### ISSUE TITLE
Searching for an integer enum value by string

### ISSUE THREAD

**Issue by Calinou** [lorisleiva/laravel-search-string#10]
In a project I'm working on, I have an [Asset model](https://github.com/Calinou/godot-asset-library-laravel/blob/master/app/Asset.php) with a `support_level_id` field, which can be set to the following values:

- `0`: Testing
- `1`: Community
- `2`: Official

I'd like to search for assets with a specific support level by writing something like `support_level:official`. However, since this package can't search in computed properties, I can only search by ID using `support_level_id:2`. (In that model, I have a `getSupportLevelAttribute()` function that returns the support's level name for use in Blade templates.)

Is there a way to implement this somehow? This is quite low-priority, so I could live without it. It'd just make the search string syntax more complete for my use case.

PS: Thanks for creating this library, it's saving me a lot of time :slightly_smiling_face:

**Comment by lorisleiva** [comment]
Hi there 👋 

Thanks for raising this.

So, unfortunately, computed properties could not really be used for this package since the search query will be generating a database query where computed properties do not exist yet.

However, your issue could be fixed by one of the following feature request.

1. Support searching using query scopes (somehow).
2. Add a `map` option in column options that maps a value into another one.

For 2. it could look something like that:

```php
protected $searchStringColumns = [
    'support_level_id' => [
        'key' => 'support_level',
        'map' => function ($value) {
            return [
                'testing' => 0,
                'community' => 1,
                'official' => 2,
            ][$value];
        }
    ],
    // ...
];
```

And that `map` option could also support a simple array.

```php
protected $searchStringColumns = [
    'support_level_id' => [
        'key' => 'support_level',
        'map' => [
            'testing' => 0,
            'community' => 1,
            'official' => 2,
        ]
    ],
    // ...
];
```

For 1. it would be a bit more difficult to implement but if we found the right approach it could bring a lot more flexibility to the library.

I don't have time to implement such feature requests at the moment but I'll leave this open until something gets done about it. 🙂

**Comment by morloderex** [comment]
@Calinou @mbardelmeijer I went ahead and tried a simple solution for this. Where you have the ability to add a maping option in the column configuration. See https://github.com/lorisleiva/laravel-search-string/pull/13

**Comment by lorisleiva** [comment]
The second option has been implemented and is available from release [`v0.1.6`](https://github.com/lorisleiva/laravel-search-string/releases/tag/v0.1.6).

Thank you @morloderex for your contribution! 🎉
